segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessorTest.java

```diff
@@ -560,89 +446,33 @@ public void testConditionalSegmentMergeReplaceIfEquals() throws Exception {
-                new WireCommands.ConditionalAttributeUpdate(randomAttribute2, WireCommands.ConditionalAttributeUpdate.REPLACE_IF_EQUALS, 2, streamSegmentName.hashCode())
-        );
-
         // The first attempt should fail as the attribute update is not going to work.
         assertTrue(append(transactionName, 1, store));
-        processor.mergeSegments(new WireCommands.MergeSegments(2, streamSegmentName, transactionName, "", attributeUpdates));
-        order.verify(connection).send(new WireCommands.SegmentAttributeUpdated(2, false));
+        WireCommands.BatchInfo batch = new WireCommands.BatchInfo(2L, 5, false);
+        processor.mergeSegments(new WireCommands.MergeSegments(7L, streamSegmentName, transactionName, "", Optional.of(batch)));
+        order.verify(connection).send(new WireCommands.SegmentAttributeUpdated(7L, false));
 
-        // Now, set the right attributes in the parent segment.
-        processor.updateSegmentAttribute(new WireCommands.UpdateSegmentAttribute(3, streamSegmentName, randomAttribute1,
-                streamSegmentName.hashCode(), WireCommands.NULL_ATTRIBUTE_VALUE, ""));
-        order.verify(connection).send(new WireCommands.SegmentAttributeUpdated(3, true));
-        processor.updateSegmentAttribute(new WireCommands.UpdateSegmentAttribute(4, streamSegmentName, randomAttribute2,
-                streamSegmentName.hashCode(), WireCommands.NULL_ATTRIBUTE_VALUE, ""));
-        order.verify(connection).send(new WireCommands.SegmentAttributeUpdated(4, true));
+        //try merging the txn segment to an invalid target stream segment, should return nosuchsegment exception
+        String nonExistingSourceSegment = "scope/stream/nonExistingSegment";
+        processor.mergeSegments(new WireCommands.MergeSegments(12L, nonExistingSourceSegment, transactionName, "", Optional.of(batch)));
+        order.verify(connection).send(new WireCommands.NoSuchSegment(12L, nonExistingSourceSegment, "", -1));
 
         // Merge segments conditionally, now it should work.
-        processor.mergeSegments(new WireCommands.MergeSegments(5, streamSegmentName, transactionName, "", attributeUpdates));
-        order.verify(connection).send(new WireCommands.SegmentsMerged(5, streamSegmentName, transactionName, 1));
-
-        // Check the value of attributes post merge.
-        processor.getSegmentAttribute(new WireCommands.GetSegmentAttribute(6, streamSegmentName, randomAttribute1, ""));
-        order.verify(connection).send(new WireCommands.SegmentAttribute(6, 1));
-        processor.getSegmentAttribute(new WireCommands.GetSegmentAttribute(7, streamSegmentName, randomAttribute2, ""));
-        order.verify(connection).send(new WireCommands.SegmentAttribute(7, 2));
-    }
-
-    @Test(timeout = 20000)
-    public void testConditionalSegmentMergeReplace() throws Exception {
```

Comment:
If you are supposed to remove the work in PR #6138, please make sure to do that completely (i.e., there is no code left from that which is not going to be used).

Reply:
That test is removed.